api/endpoint-tests/states/put.js

```diff
@@ -0,0 +1,213 @@
+const tap = require('tap'); // eslint-disable-line import/no-extraneous-dependencies
+const {
+  db,
+  getFullPath,
+  login,
+  request,
+  unauthenticatedTest,
+  unauthorizedTest
+} = require('../utils');
+
+const invalidScenarios = [
+  {
+    name: 'with an empty medicaid office',
+    token: 'edit-state-invalid-medicaid-office',
+    body: {
+      medicaid_office: {}
+    }
+  },
+  {
+    name: 'with a medicaid office with an invalid address',
+    token: 'edit-state-invalid-medicaid-office',
+    body: {
+      medicaid_office: {
+        address: 0xdeadbeef,
+        city: 'city',
+        zip: 'zip'
+      }
+    }
+  },
+  {
+    name: 'with a medicaid office with an invalid city',
+    token: 'edit-state-invalid-medicaid-office',
+    body: {
+      medicaid_office: {
+        address: 'address',
+        city: 0xdeadbeef,
+        zip: 'zip'
+      }
+    }
+  },
+  {
+    name: 'with a medicaid office with an invalid zip',
+    token: 'edit-state-invalid-medicaid-office',
+    body: {
+      medicaid_office: {
+        address: 'address',
+        city: 'city',
+        zip: 0xdeadbeef
+      }
+    }
+  },
+  {
+    name: 'with an invalid program benefits statement',
+    token: 'edit-state-invalid-benefits',
+    body: {
+      program_benefits: 0xdeadbeef
+    }
+  },
+  {
+    name: 'with an invalid program vision statement',
+    token: 'edit-state-invalid-vision',
+    body: {
+      program_vision: 0xdeadbeef
+    }
+  },
+  {
+    name: 'with points of contact with invalid name',
+    token: 'edit-state-invalid-state-pocs',
+    body: {
+      state_pocs: [
+        {
+          name: 0xdeadbeef,
+          email: 'em@il',
+          position: 'position'
+        }
+      ]
+    }
+  },
+  {
+    name: 'with points of contact with invalid email',
+    token: 'edit-state-invalid-state-pocs',
+    body: {
+      state_pocs: [
+        {
+          name: 'name',
+          email: 'email',
+          position: 'position'
+        }
+      ]
+    }
+  },
+  {
+    name: 'with points of contact with invalid position',
+    token: 'edit-state-invalid-state-pocs',
+    body: {
+      state_pocs: [
+        {
+          name: 'name',
+          email: 'em@il',
+          position: 0xdeadbeef
+        }
+      ]
+    }
+  }
+];
+
```

Comment:
Since the two endpoints (PUT `/state` and `/state/:id`) actually do the same thing after they figure out which state is the target, a lot of the tests are shared between them.  :tada: